### PR TITLE
ARROW-7910: [C++] Add internal GetPageSize() function

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

**Comment by pitrou** [comment]
@kiszk You might check this.

**Comment by pitrou** [comment]
Well, the ARM and s390x tests succeeded at least.

**Comment by kiszk** [comment]
Let me run on ppc64.

**Comment by kiszk** [comment]
Succeeded on ppc64 (big-endian), too.
```
[----------] 1 test from GetPageSize
[ RUN      ] GetPageSize.Basics
[       OK ] GetPageSize.Basics (0 ms)
[----------] 1 test from GetPageSize (0 ms total)

```

**Comment by kiszk** [comment]
LGTM

File: cpp/src/arrow/util/io_util.cc [hunk 1]
```diff
@@ -979,6 +979,32 @@ static Status StatusFromMmapErrno(const char* prefix) {
...
+  return getpagesize();
+#elif defined(_WIN32)
+  SYSTEM_INFO si;
+  GetSystemInfo(&si);
+  return si.dwPageSize;
+#else
+  errno = 0;
+  const auto ret = sysconf(_SC_PAGESIZE);
+  if (ret == -1) {
+    ARROW_LOG(FATAL) << "sysconf(_SC_PAGESIZE) failed: " << ErrnoMessage(errno);
```
Comment (fsaintjacques) [inline cpp/src/arrow/util/io_util.cc:995]: Just return 4k?

**Review by fsaintjacques — APPROVED** [review]

File: cpp/src/arrow/util/io_util.cc (hunk 1 above)
Comment (pitrou) [reply cpp/src/arrow/util/io_util.cc:995]: I'd rather know in which conditions this syscall fails (it should never).

**Pull request closed without merging**